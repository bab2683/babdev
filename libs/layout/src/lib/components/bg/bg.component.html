<div
  class="bg"
  [ngStyle]="{
    'background-image': 'url(' + path + ')',
    'background-position-x': positionX,
    'background-position-y': positionY,
    'justify-content': horizontalContentAlignment
  }"
>
  <div class="bg__content">
    <ng-content></ng-content>
  </div>
</div>
