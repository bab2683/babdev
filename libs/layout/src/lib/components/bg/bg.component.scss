@import "~@styleguide_funcional";

@mixin absolute-full {
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

:host {
  display: block;
  position: relative;
  width: 100%;
}

.bg {
  align-items: center;
  background-attachment: fixed;
  background-size: cover;
  display: flex;

  @include absolute-full;

  &::before {
    background-color: color("main");
    content: "";
    opacity: 0.7;

    @include absolute-full;
  }

  &__content {
    position: relative;
  }
}
