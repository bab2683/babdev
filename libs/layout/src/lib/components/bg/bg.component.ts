import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';

import {
  BackGroundPositionXEnum,
  BackGroundPositionYEnum,
  HorizontalContentAlignmentEnum
} from '../../enums';
import { BgModel } from '../../models';

@Component({
  selector: 'babdev-bg',
  styleUrls: ['./bg.component.scss'],
  templateUrl: './bg.component.html'
})
export class BgComponent implements OnChanges {
  @Input() public data: BgModel;
  @Input() public horizontalContentAlignment: HorizontalContentAlignmentEnum =
    HorizontalContentAlignmentEnum.CENTER;

  public path: string;
  public positionX: BackGroundPositionXEnum | string;
  public positionY: BackGroundPositionYEnum | string;

  public ngOnChanges(changes: SimpleChanges): void {
    if (changes.data) {
      const {
        path,
        positionX = BackGroundPositionXEnum.CENTER,
        positionY = 0
      } = this.data;

      this.path = path;
      this.positionX =
        typeof positionX === 'number' ? `${positionX}%` : positionX;
      this.positionY =
        typeof positionY === 'number' ? `${positionY}%` : positionY;
    }
  }
}
