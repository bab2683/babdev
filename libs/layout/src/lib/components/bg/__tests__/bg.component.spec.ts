import { SimpleChange } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import {
  BackGroundPositionXEnum,
  BackGroundPositionYEnum,
  HorizontalContentAlignmentEnum
} from '../../../enums';
import { BgComponent } from '../bg.component';

describe('BgComponent', () => {
  let component: BgComponent;
  let fixture: ComponentFixture<BgComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [BgComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BgComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should put the background', () => {
    component.data = {
      path: '/assets/contacts.jpg',
      positionY: BackGroundPositionYEnum.BOTTOM,
      positionX: 50
    };

    component.ngOnChanges({
      data: new SimpleChange(null, component.data, false)
    });

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
    expect(component.positionX).toEqual('50%');
    expect(component.positionY).toEqual(BackGroundPositionYEnum.BOTTOM);
  });

  it('should use the default values the background', () => {
    component.data = {
      path: '/assets/contacts.jpg'
    };

    component.ngOnChanges({
      data: new SimpleChange(null, component.data, false)
    });

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
    expect(component.positionX).toEqual(BackGroundPositionXEnum.CENTER);
    expect(component.positionY).toEqual('0%');
  });

  it('should not load the background', () => {
    component.horizontalContentAlignment = HorizontalContentAlignmentEnum.LEFT;

    component.ngOnChanges({
      horizontalContentAlignment: new SimpleChange(
        null,
        component.horizontalContentAlignment,
        true
      )
    });

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
