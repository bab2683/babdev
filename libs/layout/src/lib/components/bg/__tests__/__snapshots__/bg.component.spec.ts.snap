// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BgComponent should not load the background 1`] = `
<babdev-bg
  horizontalContentAlignment={[Function String]}
>
  <div
    class="bg"
    style="background-image: url(undefined); justify-content: flex-end;"
  >
    <div
      class="bg__content"
    />
  </div>
</babdev-bg>
`;

exports[`BgComponent should put the background 1`] = `
<babdev-bg
  data={[Function Object]}
  horizontalContentAlignment={[Function String]}
  path={[Function String]}
  positionX={[Function String]}
  positionY={[Function String]}
>
  <div
    class="bg"
    style="background-image: url(/assets/contacts.jpg); justify-content: center; background-position-x: 50%; background-position-y: bottom;"
  >
    <div
      class="bg__content"
    />
  </div>
</babdev-bg>
`;

exports[`BgComponent should use the default values the background 1`] = `
<babdev-bg
  data={[Function Object]}
  horizontalContentAlignment={[Function String]}
  path={[Function String]}
  positionX={[Function String]}
  positionY={[Function String]}
>
  <div
    class="bg"
    style="background-image: url(/assets/contacts.jpg); justify-content: center; background-position-x: center; background-position-y: 0%;"
  >
    <div
      class="bg__content"
    />
  </div>
</babdev-bg>
`;
